packages/java-edition/src/mcfunction/node/argument.ts

```diff
@@ -279,22 +279,67 @@ export interface FloatRangeNode extends core.AstNode {
 	value: [number | undefined, number | undefined]
 }
 
-export interface ItemNode extends core.AstNode {
+export interface ItemOldNode extends core.AstNode {
 	type: 'mcfunction:item'
 	children: (core.ResourceLocationNode | nbt.NbtCompoundNode)[]
 	id: core.ResourceLocationNode
 	nbt?: nbt.NbtCompoundNode
 }
+export namespace ItemOldNode {
+	export function is(node: core.AstNode | undefined): node is ItemOldNode {
+		return (node as ItemOldNode | undefined)?.type === 'mcfunction:item' &&
+			'nbt' in (node as ItemNode)
+	}
+}
+
+export interface ItemNewNode extends core.AstNode {
+	type: 'mcfunction:item'
+	children: (
+		| core.LiteralNode
+		| core.ResourceLocationNode
+		| ComponentListNode
+		| ComponentPredicatesNode
+	)[]
+	id: core.ResourceLocationNode
+	components?: ComponentListNode
+	componentPredicates?: ComponentPredicatesNode
+	wildcard?: boolean
+}
+
+export namespace ItemNewNode {
+	export function is(node: core.AstNode | undefined): node is ItemNewNode {
+		return (node as ItemNewNode | undefined)?.type === 'mcfunction:item' &&
+			'components' in (node as ItemNode)
```

Comment:
It this is a item predicate node `components` won't exist, so this check won't work as intended. This shouldn't be needed if we separate the `type` for `ItemOldNode` and `ItemNewNode` though as mentioned in earlier comment.